feat: use sessiond mconfig to initialize sentry in sessiond

waqaraqeel opened the pull request:
Signed-off-by: Waqar Aqeel <waqeel@fb.com>

## Summary
Working towards #7613.

This PR makes changes to sessiond sentry initialization so that relevant configuration sent from orc8r are used. Configurations in `control_proxy.yml` still take precedence over those sent from orc8r.

## Test Plan

Ran `make test_session_manager` in `lte/gateway` with `SENTRY_ENABLED` flag set.

waqaraqeel committed:
feat[orc8r]: Add generated files for sentry.io config edits

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel committed:
feat[orc8r]: move sentryconfig inside network

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel committed:
fix[orc8r]: change builder_servicer test code to pass after adding sentryconfig to sessiond

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel committed:
feat[orc8r]: add sentryconfig to mme service

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel committed:
feat[orc8r][lte]: API models to service mconfig translation and tests

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel committed:
refactor: update go-openapi/swag and package hashes

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel committed:
feat: use url_native from mconfig

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel committed:
feat: use sample_rate from mconfig

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel committed:
fix: added generated files after rebase

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

themarwhal reviewed (commented):
Can we modify `lte/gateway/configs/control_proxy.yml` so that the default sample rate is 1?

themarwhal commented on lte/gateway/c/session_manager/SentryWrappers.cpp:
```diff
@@ -37,33 +38,56 @@ using std::experimental::optional;
+    YAML::Node control_proxy_config) {
   if (control_proxy_config[SENTRY_NATIVE_URL].IsDefined()) {
-    const std::string sentry_dns =
+    const std::string override =
```
`dns_override` just to be extra clear

themarwhal commented on lte/gateway/c/session_manager/SentryWrappers.cpp:
```diff
@@ -37,33 +38,56 @@ using std::experimental::optional;
+    const magma::mconfig::SentryConfig& sentry_config,
+    YAML::Node control_proxy_config) {
+  if (control_proxy_config[SENTRY_SAMPLE_RATE].IsDefined()) {
+    const float override = control_proxy_config[SENTRY_SAMPLE_RATE].as<float>();
```
`sample_rate_override`

themarwhal reviewed (approved):
the changes look good thanks!

My only concern is that the sample rate from the config is 0.5 by default at the moment, we might want that value to be consistent with the default value we use here.

waqaraqeel committed:
fix: change default sample rate to 1.0

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel committed:
refactor: better local variable names in sentry config

Signed-off-by: Waqar Aqeel <waqeel@fb.com>

waqaraqeel commented:
> the changes look good thanks!
> 
> My only concern is that the sample rate from the config is 0.5 by default at the moment, we might want that value to be consistent with the default value we use here.

I have changed the default `sample_rate` in `control_proxy.yml` to 1.0. Also changed the local variable names to reflect your suggestions. Thanks!
